CONTRIBUTE.md

```diff
@@ -1,11 +1,109 @@
-TODO: make that a proper document
-* use Machete to remove unused dependencies
-    * cargo install cargo-machete
-    * cargo machete --with-metadata
-    * cargo machete fix
-    
-The tool is not perfect, so in case of false positive, add an entry to the Cargo.toml crate:
+# Contributing to Copper-rs
+
+First off, thank you for considering contributing to Copper-rs! We welcome contributions from everyone. This document provides guidelines for contributing to the project.
+
+## Getting Started
+
+### Prerequisites
+
+* **Rust:** Ensure you have a recent stable Rust toolchain installed. You can install it using [rustup](https://rustup.rs/).
+    ```bash
+    curl --proto '=https' --tlsv1.2 -sSf [https://sh.rustup.rs](https://sh.rustup.rs) | sh
+    ```
+* **cargo-nextest:** For running tests efficiently.
+    ```bash
+    cargo install cargo-nextest
+    ```
+* **Platform-Specific Dependencies:** Depending on your operating system and the features you intend to work with, you might need additional dependencies. Refer to the [Continuous Integration Setup](#continuous-integration-ci) section below for details extracted from our CI workflow. For Ubuntu 22.04, please checkout our [Dockerfile](support/docker/Dockerfile).
+
+
+### Building the Project
+
+To build the project, navigate to the root directory and run:
+
+```bash
+cargo build --workspace
+```
+
+For a release build, use:
+
+```bash
+cargo build --release --workspace
+```
+
+### Running Tests
+
+We use cargo-nextest for running tests. To run all unit tests:
+
+```bash
+cargo nextest run --workspace --all-targets
+```
+
+To run tests including specific features (matching the CI 'debug' mode non-CUDA features):
+
+```bash
+cargo nextest run --workspace --all-targets --features macro_debug,mock,perf-ui,image,kornia,python,gst,faer,nalgebra,glam,debug_pane,bincode
+```
+
+## Contribution Workflow
+
+1. **Fork the Repository**: Create your own fork of the copper-rs repository on GitHub.
```

Comment:
![image](https://github.com/user-attachments/assets/3d2fff46-2d5a-4ab5-9a83-9623631ff388)
^^ maybe add a little screenshot of the fork button

Reply:
This may be too much 😅